tv_apps/fling-player/test/unit/connector_test.js

```diff
@@ -0,0 +1,335 @@
+/* global Connector, castingMsgTemplate, MockPresentation */
+'use strict';
+
+requireApp('fling-player/test/unit/casting_message_template.js');
+requireApp('fling-player/test/unit/mock_presentation.js');
+requireApp('fling-player/bower_components/evt/index.js');
+requireApp('fling-player/js/casting_message.js');
+requireApp('fling-player/js/master_debug.js');
+requireApp('fling-player/js/connector.js');
+
+suite('fling-player/Connector', function() {
+
+  var exp, msg, spy, connector, presentation;
+
+  setup(function (done) {
+    presentation = new MockPresentation();
+    connector = new Connector(presentation);
```

Comment:
Generally we replace `navigator.mozPresentation` in `setup()` and put it back in `teardown()` (Instead of making a default API as a parameter).